lua/ui/game/announcement/AbstractAnnouncement.lua

```diff
@@ -0,0 +1,421 @@
+--** copies of the Software, and to permit persons to whom the Software is
+--** furnished to do so, subject to the following conditions:
+--**
+--** The above copyright notice and this permission notice shall be included in all
+--** copies or substantial portions of the Software.
+--**
+--** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+--** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+--** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+--** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+--** SOFTWARE.
+--******************************************************************************************************
+
+local LayoutHelpers = import("/lua/maui/layouthelpers.lua")
+local Layouter = LayoutHelpers.ReusedLayoutFor
+
+local UIUtil = import("/lua/ui/uiutil.lua")
+local Group = import("/lua/maui/group.lua").Group
+local Bitmap = import("/lua/maui/bitmap.lua").Bitmap
+
+local CreateLazyVar = import("/lua/lazyvar.lua").Create
+
+--- An abstract announcement that has all the functionality that you would expect from an announcement. Do not create an instance of this class.
+---@class UIAbstractAnnouncement : Destroyable, Group
+---@field Trash TrashBag
+---@field AnimateBackgroundThreadInstance? thread
+---@field AnimateThreadInstance? thread
+---@field AnimateContentThreadInstance? thread
+---@field AbortAnimationThreadInstance? thread
+---@field Background Bitmap                     # Background that we animate
+---@field BackgroundBorderTopLeft Bitmap        # Border element of the background
+---@field BackgroundBorderTop Bitmap            # Border element of the background
+---@field BackgroundBorderTopRight Bitmap       # Border element of the background
+---@field BackgroundBorderLeft Bitmap           # Border element of the background
+---@field BackgroundBorderRight Bitmap          # Border element of the background
+---@field BackgroundBorderBottomLeft Bitmap     # Border element of the background
+---@field BackgroundBorderBottom Bitmap         # Border element of the background
+---@field BackgroundBorderBottomRight Bitmap    # Border element of the background
+---@field ContentArea Group                     # Content area that we animate to/from
+AbstractAnnouncement = ClassUI(Group) {
+
+    -- Announcements works by morphing the background from a control, to the content are and back. By doing so,
+    -- we give the player an idea what the announcement is connected to. The content of the announcement is never 
+    -- moved, only the background is. The alpha of the content is adjusted when the announcement arrives and 
+    -- leaves again.
+
+    -- To make it more concrete:
+    -- - The 'ContentArea' is the area that we animate to/from
```

Comment:
to/from is easy to confuse as the origin control, which would be very similarly worded with from/to.
```suggestion
    -- - The 'ContentArea' is the area that we animate towards when displaying the announcement, and away from when hiding it
```